documentation/src/main/asciidoc/user_guide/query.adoc

```diff
@@ -92,73 +144,468 @@ for (Object book : objectList) {
+has an adverse effect on query latency.
+
+
+===== Non-shared indexes [[query.non-shared-index]]
+
+Non-shared indexes are independent indexes at each node. This setup is particularly advantageous for replicated caches where each node has all the cluster data
+and thus can hold all the indexes as well, offering optimal query performance with zero network latency when querying. Another advantage is, since the index
+is local to each node, there is less contention during writes due to the fact that each node is subjected to its own index lock, not a cluster wide one.
+
+Since each node might hold a partial index, it may be necessary to <<query.clustered-query-api, broadcast>> queries in order to get correct search results, which
+can add latency. If the cache is REPL, though, the broadcast is not necessary: each node can hold a full local copy of the index and queries runs at optimal speed
+taking advantage of a local index.
+
+Infinispan has two index managers suitable for non-shared indexes: `directory-based` and `near-real-time`. Storage wise, non-shared indexes can be located
+in ram, filesystem, or Infinispan local caches.
+
+====== Effect of the index mode
+
+The `directory-based` and `near-real-time` index managers can be associated with different <<query.index-mode,index modes>>, resulting in different index distributions.
+
+REPL caches combined with the `ALL` index mode will result in a full copy of the cluster-wide index on each node. This mode allows queries to become effectively local without
+network latency. This is the recommended mode to index any REPL cache, and that's the choice picked by the <<query.autoconfig, auto-config>> when the a REPL cache is detected. The `ALL` mode should not be used with DIST caches.
+
+REPL or DIST caches combined with `LOCAL` index mode will cause each node to index only data inserted from the same JVM, causing an uneven distribution of the index. In order
+to obtain correct query results, it's necessary to use <<query.clustered-query-api,broadcast>> queries.
+
+REPL or DIST caches combined with `PRIMARY_OWNER` will also need broadcast queries. Differently from the `LOCAL` mode, each node's index will contain indexed entries which key is primarily owned by the node according to the consistent hash, leading to a more evenly distributed indexes among the nodes.
+
+
+====== directory-based index manager [[query.directory-based]]
+
+This is the default Index Manager used when no index manager is configured. The `directory-based` index manager is used to manage indexes backed by a local lucene directory. It supports _ram_, _filesystem_ and non-clustered
+_infinispan_ storage.
+
+
+[small]*Filesystem storage*
+
+This is the default storage, and used when index manager configuration is omitted. The index is stored in the filesystem using a link:https://lucene.apache.org/core/6_0_0/core/org/apache/lucene/store/MMapDirectory.html[MMapDirectory].
```

Comment:
the link refers to Lucene 6 .. I doubt it's different but it might confuse someone?
Maybe we could inject the version as a build property? (drawback is that the links might break in future w/o us noticing)